react-components/src/components/Architecture/pointsOfInterest/PoiList.tsx

```diff
@@ -1,44 +1,61 @@
 /*!
  * Copyright 2024 Cognite AS
  */
-import { type ReactNode, useMemo, useState } from 'react';
+import { type ReactNode, useCallback, useMemo, useState } from 'react';
 import { type PointOfInterest } from '../../../architecture/concrete/pointsOfInterest/types';
 import { useOnUpdateDomainObject } from '../useOnUpdate';
 import { DataGrid, type DatagridColumn } from '@cognite/cogs-lab';
 import { usePoiDomainObject } from './usePoiDomainObject';
 import { EMPTY_ARRAY } from '../../../utilities/constants';
 import { useTranslation } from '../../i18n/I18n';
+import { Button, Flex } from '@cognite/cogs.js';
+import { take } from 'lodash';
 
 type RowType = {
   id: string;
   poi: PointOfInterest<unknown>;
 };
 
+export type PoiFilter = { contains: string };
+
 export type PoiListProps = {
   onRowClick?: (poi: PointOfInterest<unknown>) => void;
+  filter?: PoiFilter;
+  pageLimit?: number;
 };
 
-export const PoiList = ({ onRowClick }: PoiListProps): ReactNode => {
+export const PoiList = ({
+  onRowClick,
+  filter,
+  pageLimit: inputPageLimit
+}: PoiListProps): ReactNode => {
   const { t } = useTranslation();
 
+  const pageLimit = inputPageLimit ?? Infinity;
+
   const [pois, setPois] = useState<Array<PointOfInterest<unknown>>>([]);
   const [selectedPoi, setSelectedPoi] = useState<PointOfInterest<unknown> | undefined>(undefined);
 
   const poiObject = usePoiDomainObject();
 
+  const [currentLimit, setCurrentLimit] = useState<number>(pageLimit ?? Infinity);
+
   useOnUpdateDomainObject(poiObject, () => {
     setPois([...(poiObject?.pointsOfInterest ?? EMPTY_ARRAY)]);
     setSelectedPoi(poiObject?.selectedPointsOfInterest);
   });
 
+  const relevantPois = useMemo(() => pois.filter((poi) => filterPoi(poi, filter)), [pois, filter]);
```

Comment:
I feel this should be `useQuery`, what do you think

Reply:
I don't think so - this is not an async call, it just filters the PoIs that have already been fetched